lib/classes/Smtp.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Trendwerk\Smtp;
+
+final class Smtp
+{
+    public function __construct()
```

Comment:
Maybe you could create an `init` method that initializes the hooks, instead of doing this on `__construct()`. I would not recommend self-initialization for (WordPress) classes and it seems like an easy win.